velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -993,12 +993,20 @@ void updateWriterOptionsFromHiveConfig(
     const std::shared_ptr<const HiveConfig>& hiveConfig,
     const config::ConfigBase* sessionProperties,
     std::shared_ptr<dwio::common::WriterOptions>& writerOptions) {
-  if (fileFormat == dwio::common::FileFormat::PARQUET) {
+  switch (fileFormat) {
```

Comment:
@Yuhta should we make this a virtual call to remove the switch?

Reply:
It's not possible because the virtual call would be in file readers/writers and we need the Hive information here.  File formats cannot depend on Hive but the reverse is good.

Reply:
@Yuhta  Is there an actual dependency on Hive in these functions? It seems like all they are doing is fetching configs, and the parsing and fetching of these configs probably belongs with the file writer library itself. Similar to what we did for other parquet configs recently.

Reply:
It seems our change to `WriterOptions` is refactored in the mentioned PR.  Let's merge this to unblock clients and address the dispatch problem in subsequent diff.  One possible solution is to move only the Hive config values (this should be only one header file and dependency free) into `dwio/common`, since `dw` here is almost equivalent to Hive.  CC: @xiaoxmeng